.github/workflows/ci.yml

```diff
@@ -115,6 +115,15 @@ jobs:
       helm-test-packages: ${{ steps.selective-checks.outputs.helm-test-packages }}
       debug-resources: ${{ steps.selective-checks.outputs.debug-resources }}
       runs-on: ${{ steps.selective-checks.outputs.runs-on }}
+      runs-on-small: >-
+        ${{ steps.selective-checks.outputs.runs-on == 'self-hosted'
```

Comment:
I have bad experiences with conditions embedded using the expreession language. It's prone to errors. Would not it be better to generate `runs-on-small/medium/large` as direct output of selective checks? 

That has the potential that we can also base it on labels later when we connect ARC - then we could for example apply an "ARC" label to PR to test it on the ARC setup. 

This has the nice property that selective checks are also very well covered by unit tests so we can easily test any logic we will apply there.

Reply:
Suggestion: Maybe the output should be named "selective-checks.outputs.small-runner", "selective-checks.outputs.medium-runner" etc. 

That will be a nice "abstraction layer" (selective checks based on labels will choose what the "medium" runner really is.

That will also read better:

runs-on: "${{ fromJson(needs.build-info.outputs.medium-runner) }}"